internal/mongo/repository.go

```diff
@@ -0,0 +1,37 @@
+package mongo
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/JakeHumphries/gymshark-hacker-news/internal/models"
+	"github.com/pkg/errors"
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/mongo"
+	"go.mongodb.org/mongo-driver/mongo/options"
+)
+
+// Repository is a struct that exposes the mongo client and context
+type Repository struct {
+	Client *mongo.Client
+}
+
+// SaveItem saves items to the mongo database
+func (r Repository) SaveItem(ctx context.Context, item models.Item) (*models.Item, error) {
+	database := r.Client.Database("hacker-news")
+	itemsCollection := database.Collection("items")
+
+	opts := options.Update().SetUpsert(true)
+
+	update := bson.M{
+		"$set": item,
+	}
+	_, err := itemsCollection.UpdateOne(ctx, bson.M{"id": item.Id}, update, opts)
+
+	if err != nil {
```

Comment:
Seems to be a convention in go to keep the `if err != nil` line directly below where the err was defined (i.e. remove the line between the two)